custom_enumerable_methods.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    my_self = []
+    i = 0
+    while i < my_self.size
+      yield(my_self[i])
+      i += 1
+    end
+    my_self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_self = []
+    (i...my_self.size).my_each do |i|
+      yield(my_self[i], i)
+    end
+    my_self
+  end
+
+  def my_select
+    result = []
+    my_self = []
+    my_self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?
+    result = true
+    my_self = []
+    my_self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_self = []
+    my_self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_self = []
+    my_self.my_each do |i|
+      result = false if my_self[i] == yield(i)
+    end
+    result
+  end
+
+  def my_count
+    i = 0
+    my_self = []
+    my_self.my_each { i += 1 }
+    i
+  end
```

Comment:
```rb
2.6.5 :116 > array
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :117 > array.my_count
 => 0 
2.6.5 :118 > array.count
 => 100 
2.6.5 :119 >
```

Reply:
```rb
2.6.5 :121 > array
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :122 > array.my_count(0)
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        2: from (irb):122
        1: from custom_enumerable_methods.rb:54:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :123 > array.count(0)
 => 15 
2.6.5 :124 >
```

Reply:
```rb
2.6.5 :126 > block = proc { |num| num < 4 }
2.6.5 :127 > array.my_count(&block)
 => 0 
2.6.5 :128 > array.count(&block)
 => 45 
2.6.5 :129 >
```

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more and fix the things.